BCP/Session Lifecycle Management.md

```diff
@@ -0,0 +1,146 @@
+
+## Motivation
+
+The motivation behind this proposal is to enhance the flexibility of CAIP-25 by enabling management of session authorizations without sessionIds which don't map well to extension based wallet's dapp connections and could therefore add constraints and burdens to existing flows. The proposed methods provide an intuitive way to add, revoke, and retrieve authorizations within an existing session, simplifying the management of session lifecycles.
+
+## Lifecycle Equivalence Chart
+
+||feature|CAIP-25 now w/sessionId|CAIP-285 w/o sessionId|
+|---|---|---|---|
+|1|dapp initialize (replaces session if already exist)|call `wallet_createSession` w/no sessionId |call `wallet_createSession` w/no sessionId|
+|2|wallet re-initialize|return `wallet_createSession` w/new sessionId **next time called**|n/a (not needed because [`wallet_sessionChanged`][CAIP-308] notification can be sent, and wallet_getSession can be used to confirm everything is good)|
+|3|dapp get current session|n/a (should persist)|[`wallet_getSession`][CAIP-307]  w/o sessionId|
+|4|dapp confirm current session|call `wallet_createSession` w/sessionId and same properties OR `wallet_getSession` w/sessionId|`wallet_getSession` w/o sessionId|
+|5|dapp revoke|call `wallet_createSession` w/no sessionId and no scopes OR [`wallet_revokeSession`][CAIP-285] w/sessionId |`wallet_revokeSession`  w/o sessionId|
+|6|wallet revoke|return `wallet_createSession` w/new sessionId and no scopes **next time called** or `wallet_sessionChanged` w/ sessionId |`wallet_sessionChanged`  w/no scopes|
+|7|dapp update session|call `wallet_createSession` w/existing sessionId and new scopes|call `wallet_createSession` w/no sessionId|
+|8|wallet update session|return `wallet_createSession` w/new sessionId and no scopes **next time called** OR `wallet_sessionChanged` w/existing sessionId|`wallet_sessionChanged` w/o sessionId|
+
+## Lifecycle diagrams
+
+### Visualizing the lifecycle of a session without a `sessionId`
+
+```mermaid
+sequenceDiagram
+  rect rgb(255, 255, 255)
+  actor Caller
+  participant Wallet
+  participant WalletDataStore as Data Store (Wallet)
+  rect rgb(255, 245, 245)
+  note right of Caller: Session Creation
+  Caller->>Wallet: wallet_createSession
+  Wallet->>WalletDataStore: Persist session data
+  Wallet-->>Caller: {"sessionScopes": {...}}
+  end
+  rect rgb(245, 255, 245)
+  note right of Caller: Update Session
+  Caller->>Wallet: wallet_createSession (update auth)
+  Wallet->>WalletDataStore: Update session data
+  Wallet-->>Caller: {"sessionScopes": {updatedScopes...}}
+  end
+  rect rgb(245, 245, 255)
+  note right of Caller: Connection Interrupted w/ Wallet Side Session Modification
+  Caller-->Wallet: Connection Interrupted
+  Wallet->>WalletDataStore: User initiated session change
+  Wallet->>Caller: wallet_sessionChanged
```

Comment:
this notification can't be sent since connection is interrupted

Reply:
Yeah I think the idea here is that it attempts to send it but its not received. Will see if there is a better way to represent this in the diagram